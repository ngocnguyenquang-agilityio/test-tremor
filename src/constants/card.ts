// Profile Contact Card
export const INFO_LIST = [
  {
    information: "Full Name:",
    description: "Alec M. Thompson",
  },
  {
    information: "Mobile:",
    description: "(44) 123 1234 123",
  },
  {
    information: "Email:",
    description: "alecthompson@mail.com",
  },
  {
    information: "Location:",
    description: "USA",
  },
];
