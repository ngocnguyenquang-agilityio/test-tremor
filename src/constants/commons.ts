export const ITEM_ACTION_PROJECT = [
  {
    key: "act",
    label: "Action",
  },
  {
    key: "anotherAct",
    label: "Another action",
  },
  {
    key: "somethingElse",
    label: "Something else here",
  },
];

export const URL_AVATAR_PROJECT = [
  {
    key: "avatar1",
    src: "/images/avatar/avatar-xs.webp",
  },
  {
    key: "avatar2",
    src: "/images/avatar/avatar-xs.webp",
  },
  {
    key: "avatar3",
    src: "/images/avatar/avatar-xs.webp",
  },
];
