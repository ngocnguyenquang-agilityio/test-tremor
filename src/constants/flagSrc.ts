type FlagType = {
  [key: string]: string;
  us: string;
  ge: string;
  gb: string;
  br: string;
};

export const FLAG_SRC: FlagType = {
  us: "/assets/images/flags/US.webp",
  ge: "/assets/images/flags/DE.webp",
  gb: "/assets/images/flags/GB.webp",
  br: "/assets/images/flags/BR.webp",
};
