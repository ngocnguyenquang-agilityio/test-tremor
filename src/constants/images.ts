export const NOT_FOUND = {
  SRC: "/images/not-found.png",
};
