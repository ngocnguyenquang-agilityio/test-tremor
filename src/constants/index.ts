export * from "./routes";
export * from "./regex";
export * from "./messages";
export * from "./profile";
