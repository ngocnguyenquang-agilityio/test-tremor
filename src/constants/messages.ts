export const MESSAGES_ERROR = {
  NAME_MIN_LENGTH: "Minimum length should be 4.",
  NAME_REQUIRED: "Name is required.",
  PASSWORD_REQUIRED: "Password is required.",
  EMAIL_REQUIRED: "Email is required.",
  EMAIL_INVALID: "Invalid email address.",
  PASSWORD_INVALID:
    "Password must have at least 4 characters, 1 special character and 1 number",
};
