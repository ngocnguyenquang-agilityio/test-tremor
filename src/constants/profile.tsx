import { MdEmail, MdHome, MdSettings } from "react-icons/md";

export const TABS_HEADER = [
  { name: "App", icon: <MdHome /> },
  { name: "Message", icon: <MdEmail /> },
  { name: "Setting", icon: <MdSettings /> },
];
