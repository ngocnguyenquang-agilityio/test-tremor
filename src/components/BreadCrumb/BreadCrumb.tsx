// Libs
import { memo } from "react";
import Link from "next/link";
import { Title } from "@tremor/react";
import { MdHome } from "react-icons/md";

// Constants
import { ROUTES } from "../../constants";

// Types
import { IBreadCrumb } from "@/types";

// Define the props for the BreadCrumb component
interface IBreadCrumbProps {
  links: IBreadCrumb[];
  title: string;
}

/**
 * Primary UI component for BreadCrumb component
 */
const BreadCrumb = ({ links, title }: IBreadCrumbProps): JSX.Element => {
  const renderLinks = (): JSX.Element[] => {
    return links.map(link => (
      <li key={link.id} className="flex items-center">
        <div className="bc-link">
          <Link className="text-sm capitalize" href={link.url}>
            {link.name}
          </Link>
        </div>
        <div aria-hidden="true" className="text-xs mx-2">
          &#8260;
        </div>
      </li>
    ));
  };

  return (
    <nav className="block">
      <ol className="flex flex-wrap items-center text-gray-400">
        <li>
          <Link href={ROUTES.HOME} className="flex">
            <MdHome className="bg-inherit border-0 py-0 hover:bg-transparent" />
          </Link>
        </li>
        <li aria-hidden="true" className="text-xs mx-2">
          &#8260;
        </li>

        {renderLinks()}

        <li className="text-sm text-tremor-content-title capitalize">
          {title}
        </li>
      </ol>
      <Title className="text-tremor-content-title font-bold capitalize">
        {title}
      </Title>
    </nav>
  );
};

export default memo(BreadCrumb);
