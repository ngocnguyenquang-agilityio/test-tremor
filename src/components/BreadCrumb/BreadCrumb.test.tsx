import { render } from "@testing-library/react";

// Components
import BreadCrumb from "./BreadCrumb";

// Constants
import { ROUTES } from "@/constants";

describe("Testing breadcrumb component", () => {
  const propsDefault = {
    links: [
      {
        id: "1",
        name: "dashboards",
        url: `${ROUTES.ANALYTICS}`,
      },
    ],
    title: "analytics",
  };

  it("should render correctly with display breadcrumb text", () => {
    const { container } = render(<BreadCrumb {...propsDefault} />);
    const link = container.getElementsByClassName("bc-link")[0];
    expect(propsDefault.links[0].name).toEqual("dashboards");
  });

  it("Should match snapshot", () => {
    const { container } = render(<BreadCrumb {...propsDefault} />);
    expect(container).toMatchSnapshot();
  });
});
