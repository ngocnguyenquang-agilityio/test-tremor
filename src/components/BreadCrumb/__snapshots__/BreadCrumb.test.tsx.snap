// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing breadcrumb component Should match snapshot 1`] = `
<div>
  <nav
    class="block"
  >
    <ol
      class="flex flex-wrap items-center text-gray-400"
    >
      <li>
        <a
          class="flex"
          href="/dashboards/analytics"
        >
          <svg
            class="bg-inherit border-0 py-0 hover:bg-transparent text-inherit"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
            />
          </svg>
        </a>
      </li>
      <li
        aria-hidden="true"
        class="text-xs mx-2"
      >
        ⁄
      </li>
      <li
        class="flex items-center text-inherit"
      >
        <div
          class="bc-link"
        >
          <a
            class="text-sm capitalize"
            href="/dashboards/analytics"
          >
            dashboards
          </a>
        </div>
        <div
          aria-hidden="true"
          class="text-xs mx-2"
        >
          ⁄
        </div>
      </li>
      <li
        class="text-sm text-tremor-content-title capitalize text-inherit"
      >
        analytics
      </li>
    </ol>
    <p
      class="text-tremor-title dark:text-dark-tremor-content-emphasis font-bold capitalize text-primary"
    >
      analytics
    </p>
  </nav>
</div>
`;
